src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -83,6 +83,27 @@ public static void AddProject(this SlnFile slnFile, string fullProjectPath, ILis
 
                 if (solutionFolders != null)
                 {
```

Comment:
With this added code, this scope-block becomes complex enough that it should be delegated into it's own set of logic, especially considering it is easily scopeable to a single "thing." I would move it into a function called `SetupSolutionFolders`, `ValidateSolutionFolderLogic`, or something like that.